src/components/habit-tracker/analytics/month-view/LineChart.tsx

```diff
@@ -0,0 +1,193 @@
+import ReactApexChart from "react-apexcharts";
+import styled from "styled-components";
+import { Logging } from "../../../../containers/habit-tracker/habits/Habits";
+import { colors } from "../../../../common/utilities/color-utils";
+import { getDaysInMonth } from "../../../../common/utilities/habit-analytics";
+
+// Styled component for the chart container
+const ChartContainer = styled.div`
+  width: 100%;
+  height: calc(100% - 40px);
+  margin-top: 40px;
+
+  .apexcharts-canvas {
+    background: ${colors.white};
+    border-radius: 8px;
+  }
+
+  @media (max-width: 1024px) {
+    height: 30vh;
+  }
+
+  @media (max-width: 768px) {
+    height: 60vh;
+  }
+
+  @media (max-width: 480px) {
+    height: 40vh;
+  }
+`;
+
+// interface for the line chart props
+interface LineChartProps {
+  habitsLog: Logging;
+  selectedHabit: string;
+  year: number;
+  month: string;
+}
+
+// Function to generate the ApexCharts options object
+const generateChartOptions = (
+  daysInMonth: number,
+  month: string
+): ApexCharts.ApexOptions => ({
+  chart: {
+    type: "line",
+    height: "100%",
+    toolbar: {
+      show: false,
+    },
+  },
+  stroke: {
+    curve: "smooth",
+    width: 2,
+  },
+  xaxis: {
+    categories: Array.from({ length: daysInMonth }, (_, i) =>
+      (i + 1).toString()
+    ),
+    title: {
+      text: "Day of the Month",
+      style: {
+        color: colors.green,
+      },
+    },
+    labels: {
+      style: {
+        colors: colors.blue,
+        fontSize: "10px",
+      },
+      rotate: -45,
+      hideOverlappingLabels: true,
+    },
+  },
+  yaxis: {
+    title: {
+      text: "Logs",
+      style: {
+        color: colors.green,
+      },
+    },
+    labels: {
+      style: {
+        colors: colors.blue,
+      },
+      formatter: (value: number) => Math.floor(value).toString(),
+    },
+    min: 0,
+    tickAmount: 5,
+  },
+  colors: [colors.blue],
+  markers: {
+    size: 4,
+    colors: [colors.blue],
+    strokeColors: colors.white,
+    strokeWidth: 2,
+  },
+  tooltip: {
+    enabled: true,
+    x: {
+      formatter: (dayOfMonth: number) => `${month} ${dayOfMonth}`,
```

Comment:
Not a big deal, but the first letter should be capital for the month.
![image](https://github.com/user-attachments/assets/78b5e63c-09c9-4a79-908b-b179891be9c1)